npm/private/npm_translate_lock.bzl

```diff
@@ -6,203 +6,26 @@ load(":utils.bzl", "utils")
-        to a list of Bazel packages in which to hoist the package to the top-level of the node_modules tree when linking.
-
-        This is similar to setting https://pnpm.io/npmrc#public-hoist-pattern in an .npmrc file outside of Bazel, however,
-        wild-cards are not yet supported and npm_translate_lock will fail if there are multiple versions of a package that
-        are to be hoisted.""",
-    ),
-    "dev": attr.bool(
-        doc = """If true, only install devDependencies""",
-    ),
-    "no_optional": attr.bool(
-        doc = """If true, optionalDependencies are not installed""",
-    ),
-    "lifecycle_hooks_exclude": attr.string_list(
-        doc = """A list of package names or package names with their version (e.g., "my-package" or "my-package@v1.2.3")
-        to not run lifecycle hooks on""",
-    ),
-    "run_lifecycle_hooks": attr.bool(
-        doc = """If true, runs preinstall, install and postinstall lifecycle hooks on npm packages if they exist""",
-        default = True,
-    ),
-    "lifecycle_hooks_envs": attr.string_list_dict(
-        doc = """Environment variables applied to the preinstall, install and postinstall lifecycle hooks on npm packages.
-        The environment variables can be defined per package by package name or globally using "*".
-        Variables are declared as key/value pairs of the form "key=value".
-        For example:
-        lifecycle_hooks_envs: {
-            "*": ["GLOBAL_KEY1=value1", "GLOBAL_KEY2=value2"],
-            "@foo/bar": ["PREBULT_BINARY=http://downloadurl"],
-        }
-        """,
-    ),
-    "lifecycle_hooks_execution_requirements": attr.string_list_dict(
-        doc = """Execution requirements applied to the preinstall, install and postinstall lifecycle hooks on npm packages.
-        The execution requirements can be defined per package by package name or globally using "*".
-        For example:
-        lifecycle_hooks_execution_requirements: {
-            "*": ["requires-network"],
-            "@foo/bar": ["no-sandbox"],
-        }
-        """,
-    ),
-    "lifecycle_hooks_no_sandbox": attr.bool(
-        doc = """If True, a "no-sandbox" execution requirement is added to all lifecycle hooks.
-
-        Equivalent to adding `"*": ["no-sandbox"]` to lifecycle_hooks_execution_requirements.
-
-        This defaults to True to limit the overhead of sandbox creation and copying the output
-        TreeArtifacts out of the sandbox.
-        """,
-        default = True,
-    ),
-    "verify_node_modules_ignored": attr.label(
-        doc = """node_modules folders in the source tree should be ignored by Bazel.
-
-        This points to a `.bazelignore` file to verify that all nested node_modules directories
-        pnpm will create are listed.
-
-        See https://github.com/bazelbuild/bazel/issues/8106
-        """,
-    ),
-    "warn_on_unqualified_tarball_url": attr.bool(
-        default = True,
-    ),
+    "pnpm_lock": attr.label(),
```

Comment:
This means bazel has no reference to the label docs?

Reply:
Yeah. The docs have been hoisted to the macro now. Same as npm_import already does.

Reply:
Bazel no longer knows of the doc strings though. Is that ok? Does bazel use or provide those in any way?

Reply:
stardocs docs are still generated correct from the macro docstring summary and Args. Other than that the docstrings are not load bearing anywhere.